# HG changeset patch
# User Orion Poplawski <orion@cora.nwra.com>
# Date 1430511111 25200
#      Fri May 01 13:11:51 2015 -0700
# Branch stable
# Node ID 6446649e42c187be56c1084bd505e62158f2b2e9
# Parent  4a41b7d426b604ca71df2d945168b502aae68b35
Allow Qt GUI to compile on ARM CPU architectures (bug #44970).

* QtHandlesUtils.cc (toRGB): Instantiate a temporary array of qreal which is
used for the return results of getRgbF().  Copy data from qreal into octave Matrix
of base type double.

diff --git a/libgui/graphics/QtHandlesUtils.cc b/libgui/graphics/QtHandlesUtils.cc
--- a/libgui/graphics/QtHandlesUtils.cc
+++ b/libgui/graphics/QtHandlesUtils.cc
@@ -147,17 +147,21 @@ fromRgb (const Matrix& rgb)
 }
 
 Matrix
 toRgb (const QColor& c)
 {
   Matrix rgb (1, 3);
   double* rgbData = rgb.fortran_vec ();
 
-  c.getRgbF (rgbData, rgbData+1, rgbData+2);
+  // qreal is a typedef for double except for ARM CPU architectures
+  // where it is a typedef for float (Bug #44970).
+  qreal tmp[3];
+  c.getRgbF (tmp, tmp+1, tmp+2);
+  rgbData[0] = tmp[0]; rgbData[1] = tmp[1]; rgbData[2] = tmp[2];
 
   return rgb;
 }
 
 std::string
 figureSelectionType (QMouseEvent* event, bool isDoubleClick)
 {
   if (isDoubleClick)
